singularity-opac/photons/gray_opacity_photons.hpp

```diff
@@ -49,19 +49,48 @@ class GrayOpacity {
                                                    lambda);
   }
 
+  template <typename FrequencyIndexer, typename DataIndexer>
+  PORTABLE_INLINE_FUNCTION void AbsorptionCoefficientPerNu(
```

Comment:
why does this gray opacity make repeated calls to fill in opacities for different frequency bins?

Reply:
As proof of principle. It should just return rho*kappa in this case. I will optimize this later when I have non-gray opacities to work with.